### ISSUE TITLE
[gateway] Backblaze B2 HeadObject fails on 0 byte files

### ISSUE THREAD

**Issue by digitalpardoe** [minio/minio#8166]
## Expected Behavior
A `HeadObject` request is made to MinIO using the S3 API for a 0 byte file served by the Backblaze B2 gateway, response is that the file exists as expected.

## Current Behavior
The request fails with the error:

```
Error: b2_download_file_by_name: 416: byte range bytes=0-0 for a file of length 0
       4: cmd/api-errors.go:1694:cmd.toAPIErrorCode()
       3: cmd/api-errors.go:1719:cmd.toAPIError()
       3: cmd/api-errors.go:1719:cmd.toAPIError()
       1: net/http/server.go:1995:http.HandlerFunc.ServeHTTP()
```

Backblaze apparently consider the 0-0 byte range to be "out of range" for a 0 byte file, looking at their API it may be attempting to return bytes 0 to 1.

## Possible Solution
It would be less performant but `b2_get_file_info` may be a more reliable way to satisfy the `HeadObject` request.

## Steps to Reproduce (for bugs)
1. Upload 0 byte file to B2
2. Attempt to download the file from the MinIO web UI

## Context
These 0 byte files are often used as a form of lock file for backup utilities etc. that make use of the S3 API.

## Regression
This is my first attempt at using MinIO

## Your Environment
* Version used (`minio version`): `RELEASE.2019-08-29T00-25-01Z`
* Environment name and version (e.g. nginx 1.9.1): MinIO provided Docker container

**Comment by digitalpardoe** [comment]
It looks like it was possibly using the "get info" method at some point in the past then was changed to use range. I can see the "get info" code still exists in there but my Go is terrible so that's about as far as I've gotten: https://github.com/minio/minio/pull/6508/.

**Comment by kannappanr** [comment]
Closing this issue as Fixed as #8169 has been merged
